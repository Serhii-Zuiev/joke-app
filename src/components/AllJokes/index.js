export { default as AllJokesContainer } from "./AllJokesContainer";
export { default as JokesList } from "./JokesList";
export { default as JokeItem } from "./JokeItem";