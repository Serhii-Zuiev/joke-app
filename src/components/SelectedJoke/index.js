export { default as SelectedJoke } from "./SelectedJoke";