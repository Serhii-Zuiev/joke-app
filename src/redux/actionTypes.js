export const types = {
  FETCH_JOKES_REQUESTED: "FETCH_JOKES_REQUESTED",
  FETCH_JOKES_SUCCESS: "FETCH_JOKES_SUCCESS",
  FETCH_JOKES_ERROR: "FETCH_JOKES_ERROR",
  START_LOADING: "START_LOADING",
  STOP_LOADING: "STOP_LOADING",
  FILTER_JOKES: "FILTER_JOKES",
};
