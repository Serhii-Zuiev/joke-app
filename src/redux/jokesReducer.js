export const types = {
  FETCH_JOKES_REQ: "joke-app/reducer/FETCH_JOKES_REQ",
  FETCH_JOKES_OK: "joke-app/reducer/FETCH_JOKES_OK",
  FETCH_JOKES_ERR: "joke-app/reducer/FETCH_JOKES_ERR",
  FETCH_SELECTED_JOKE_REQ: "joke-app/reducer/FETCH_SELECTED_JOKE_REQ",
  START_LOADING: "joke-app/reducer/START_LOADING",
  STOP_LOADING: "joke-app/reducer/STOP_LOADING",
  FILTER_JOKES: "joke-app/reducer/FILTER_JOKES",
};



export const innitialState = {
  jokes: [],
  filter: "All",
  isLoading: false,
  error: null,
};

export default function reducer(state = innitialState, { type, payload }) {
  switch (type) {
    case types.FETCH_JOKES_OK:
      return { ...state, jokes: [...state.jokes, ...payload] };

    case types.FILTER_JOKES:
      return { ...state, filter: payload };

    case types.START_LOADING:
      return { ...state, isLoading: true };

    case types.STOP_LOADING:
      return { ...state, isLoading: false };

    case types.FETCH_JOKES_ERR:
      return { ...state, error: payload };

    default:
      return state;
  }
}



export const fetchJokesRequested = (params) => ({
  type: types.FETCH_JOKES_REQ,
  payload: params,
});

export const fetchSelectedJokeRequested = (params) => ({
  type: types.FETCH_SELECTED_JOKE_REQ,
  payload: params,
});

export const fetchJokesSuccess = (data) => ({
  type: types.FETCH_JOKES_OK,
  payload: data,
});

export const fetchJokesError = (data) => ({
  type: types.FETCH_JOKES_ERR,
  payload: data,
});

export const startLoading = () => ({
  type: types.START_LOADING,
});

export const stopLoading = () => ({
  type: types.STOP_LOADING,
});

export const filterJokes = (filter) => ({
  type: types.FILTER_JOKES,
  payload: filter,
});
