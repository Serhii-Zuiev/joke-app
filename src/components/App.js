import React from "react";
import styled from "styled-components";
import AllJokes from "./AllJokes/AllJokesContainer";
import SelectedJoke from "./SelectedJoke/SelectedJoke";

function App() {
  return (
    <StyledAppContainer>
      <StyledHeader>
        <StyledAppHeading>
          <span role="img" aria-label="face emoji">
            ðŸ˜„
          </span>
          Joke app
        </StyledAppHeading>
      </StyledHeader>

      <AllJokes />
      {/* <SelectedJoke /> */}
    </StyledAppContainer>
  );
}

export default App;

const StyledAppContainer = styled.div`
  background-color: #eee;
`;

const StyledHeader = styled.header`
  height: 70px;
  width: 90%;
  margin: 0 auto;
`;

const StyledAppHeading = styled.h1`
  font-size: 26px;
  color: #333;
  padding: 16px 0;
  letter-spacing: 1px;
  text-align: center;
  border-bottom: 2px solid #777;
`;
