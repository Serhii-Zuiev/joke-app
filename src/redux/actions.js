import { types } from "./actionTypes";

export const fetchJokesRequested = (params) => ({
  type: types.FETCH_JOKES_REQUESTED,
  payload: params,
});

export const fetchJokesSuccess = (data) => ({
  type: types.FETCH_JOKES_SUCCESS,
  payload: data,
});

export const fetchJokesError = (data) => ({
  type: types.FETCH_JOKES_ERROR,
  payload: data,
});

export const startLoading = () => ({
  type: types.START_LOADING,
});

export const stopLoading = () => ({
  type: types.STOP_LOADING,
});

export const filterJokes = (filter) => ({
  type: types.FILTER_JOKES,
  payload: filter,
});
