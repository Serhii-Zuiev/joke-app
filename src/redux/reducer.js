import { types } from "./actionTypes";

const innitialState = {
  jokes: [],
  filter: "All",
  isLoading: false,
  error: null,
};

export function reducer(state = innitialState, { type, payload }) {
  switch (type) {
    case types.FETCH_JOKES_SUCCESS:
      return { ...state, jokes: [...state.jokes, ...payload] };

    case types.FILTER_JOKES:
      return { ...state, filter: payload.filter };

    case types.START_LOADING:
      return { ...state, isLoading: true };

    case types.STOP_LOADING:
      return { ...state, isLoading: false };
    //TODO ошибка запускает луп.
    case types.FETCH_JOKES_ERROR:
      return { ...state, error: payload };

    default:
      return state;
  }
}
