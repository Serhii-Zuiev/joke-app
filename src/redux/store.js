import { createStore, compose, applyMiddleware } from "redux";
import { devToolsEnhancer } from "redux-devtools-extension";
import createSagaMiddleware from "redux-saga";
import { sagaWatcher } from "../saga/sagas";
import { reducer } from "./reducer";

const saga = createSagaMiddleware();

export const store = createStore(
  reducer,
  compose(applyMiddleware(saga), devToolsEnhancer())
);

saga.run(sagaWatcher);
