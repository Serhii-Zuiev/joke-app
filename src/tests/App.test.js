import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";

import { App } from "../components";
import { default as reducer, innitialState } from "../redux/jokesReducer";

const isLoading = { isLoading: true };

function getProviderWrapper(mockData) {
    const mockStore = createStore(reducer, mockData);
    return mount(
        <Provider store={mockStore}>
            <App />
        </Provider>
    );
}

describe("<App/> unit test", () => {
    it("should render App component", () => {
        const wrapper = getProviderWrapper(innitialState);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render routing component if not loading", () => {
        const routing = getProviderWrapper(innitialState).find("Routing");
        expect(routing.length).toBe(1);
    });

    it("should render spinner component if loading", () => {
        const spinner = getProviderWrapper(isLoading).find("LoadingSpinner");
        expect(spinner.length).toBe(1);
    });
});
