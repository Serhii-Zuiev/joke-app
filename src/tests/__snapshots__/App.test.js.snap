// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> unit test should render App component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <Memo(GlobalStyleComponent) />
    <styled.div>
      <div
        className="sc-AxjAm bDdSts"
      >
        <div
          className="appWrapper"
        >
          <styled.header>
            <header
              className="sc-AxirZ idrxSW"
            >
              <styled.h1>
                <h1
                  className="sc-AxiKw dsGgMU"
                >
                  <span
                    aria-label="face emoji"
                    role="img"
                  >
                    ðŸ˜„
                  </span>
                  Joke app
                </h1>
              </styled.h1>
            </header>
          </styled.header>
          <Routing>
            <BrowserRouter>
              <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              >
                <Switch>
                  <Route
                    component={[Function]}
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    path="/"
                  >
                    <AllJoke
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <styled.section>
                        <section
                          className="sc-AxgMl gTGhQd"
                        >
                          <styled.h2>
                            <h2
                              className="sc-AxheI etBnmE"
                            >
                              Here are our best jokes, but it isn't certain...
                            </h2>
                          </styled.h2>
                          <Form
                            inline={false}
                            style={
                              Object {
                                "width": "6rem",
                              }
                            }
                          >
                            <form
                              className=""
                              style={
                                Object {
                                  "width": "6rem",
                                }
                              }
                            >
                              <FormGroup
                                controlId="exampleForm.SelectCustom"
                              >
                                <div
                                  className="form-group"
                                >
                                  <FormLabel
                                    column={false}
                                    srOnly={false}
                                    style={
                                      Object {
                                        "width": "8rem",
                                      }
                                    }
                                  >
                                    <label
                                      className="form-label"
                                      htmlFor="exampleForm.SelectCustom"
                                      style={
                                        Object {
                                          "width": "8rem",
                                        }
                                      }
                                    >
                                      choose category
                                    </label>
                                  </FormLabel>
                                  <FormControl
                                    as="select"
                                    custom={true}
                                    onChange={[Function]}
                                    value="All"
                                  >
                                    <select
                                      className="custom-select"
                                      id="exampleForm.SelectCustom"
                                      onChange={[Function]}
                                      value="All"
                                    >
                                      <option>
                                        All
                                      </option>
                                      <option>
                                        Nerdy
                                      </option>
                                      <option>
                                        Explicit
                                      </option>
                                    </select>
                                  </FormControl>
                                </div>
                              </FormGroup>
                            </form>
                          </Form>
                          <JokesList
                            allJokes={Array []}
                          >
                            <styled.ul>
                              <ul
                                className="sc-AxmLO cZbuqf"
                              />
                            </styled.ul>
                          </JokesList>
                          <styled.form
                            onSubmit={[Function]}
                          >
                            <form
                              className="sc-Axmtr jRulXR"
                              onSubmit={[Function]}
                            >
                              <FormControl
                                max="20"
                                min="1"
                                onChange={[Function]}
                                placeholder="1-20"
                                required={true}
                                style={
                                  Object {
                                    "width": "5rem",
                                  }
                                }
                                type="number"
                                value=""
                              >
                                <input
                                  className="form-control"
                                  max="20"
                                  min="1"
                                  onChange={[Function]}
                                  placeholder="1-20"
                                  required={true}
                                  style={
                                    Object {
                                      "width": "5rem",
                                    }
                                  }
                                  type="number"
                                  value=""
                                />
                              </FormControl>
                              <Button
                                active={false}
                                disabled={false}
                                type="submit"
                                variant="primary"
                              >
                                <button
                                  className="btn btn-primary"
                                  disabled={false}
                                  type="submit"
                                >
                                  Get more
                                </button>
                              </Button>
                            </form>
                          </styled.form>
                        </section>
                      </styled.section>
                    </AllJoke>
                  </Route>
                </Switch>
              </Router>
            </BrowserRouter>
          </Routing>
        </div>
      </div>
    </styled.div>
  </App>
</Provider>
`;
